### PR TITLE
fix(Notifications) : Handle the validation of the notification name when the name already exists & fix some errors

### PR DESCRIPTION
**Pull request by Med1233**
## Description

**PLEASE MAKE SURE THAT THE BRANCH PR INCLUDES JIRA TICKET ID** (_for centreon-internal_)

Please include a short resume of the changes and what is the purpose of PR. Any relevant information should be added to help:
* **QA Team** (Quality Assurance) with tests.
* **reviewers** to understand what are the stakes of the pull request.

**Fixes** # (issue)

## Type of change

- [ ] Patch fixing an issue (non-breaking change)
- [ ] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software

## Target serie

- [ ] 21.10.x
- [ ] 22.04.x
- [ ] 22.10.x
- [ ] 23.04.x
- [ ] 23.10.x (master)

<h2> How this pull request can be tested ? </h2>

Please describe the **procedure** to verify that the goal of the PR is matched. Provide clear instructions so that it can be **correctly tested**.

Any **relevant details** of the configuration to perform the test should be added.

#### Community contributors & Centreon team

- [ ] I have followed the **coding style guidelines** provided by Centreon
- [ ] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [ ] I have commented my code, especially **hard-to-understand areas** of the PR.
- [ ] I have **rebased** my development branch on the base branch (master, maintenance).

### CODE REVIEW

**Review by cg-tw — APPROVED** [review]

**Review by a-launois — APPROVED** [review]

**Review by abuathier — DISMISSED** [review]
LGTM

File: centreon/www/front_src/src/CloudNotificationsConfiguration/Listing/index.tsx [hunk 1]
```diff
@@ -36,13 +38,29 @@ const NotificationsListing = (): JSX.Element => {
...
   const setLimit = useSetAtom(limitAtom);
   const setEditedNotificationId = useSetAtom(EditedNotificationIdAtom);
   const setPanelMode = useSetAtom(panelModeAtom);
+  const setNotificationsNames = useSetAtom(notificationsNamesAtom);
 
-  const { loading, data: listingData } = useLoadingNotifications();
+  const { loading, data: listingData, refetch } = useLoadingNotifications();
+
+  useEffect(() => {
+    if (reload === true) {
```
Comment (Thebarda) [inline centreon/www/front_src/src/CloudNotificationsConfiguration/Listing/index.tsx:52]:
```suggestion
    if (reload) {
```

Comment (Thebarda) [reply centreon/www/front_src/src/CloudNotificationsConfiguration/Listing/index.tsx:52]: Btw, why do you need a reload state to refetch a query? With react-query, you can just either invalidate the query or refetch the query

File: centreon/www/front_src/src/CloudNotificationsConfiguration/index.tsx [hunk 2]
```diff
@@ -21,27 +23,39 @@ const useStyle = makeStyles()((theme: Theme) => ({
...
     queryOptions: {
       suspense: false
     }
   });
+
+  useEffect(() => {
+    if (reload === true) {
+      refetch();
+    }
+  }, [reload]);
```
Comment (Thebarda) [inline centreon/www/front_src/src/CloudNotificationsConfiguration/index.tsx:42]: Same comment here

File: centreon/www/front_src/src/CloudNotificationsConfiguration/Listing/index.tsx (hunk 1 above)
Comment (Med1233) [reply centreon/www/front_src/src/CloudNotificationsConfiguration/Listing/index.tsx:52]: However, the reload state can serve as an event detector. In this case, it checks if the submission was successful.

Comment (Med1233) [reply centreon/www/front_src/src/CloudNotificationsConfiguration/Listing/index.tsx:52]: what do you think @Thebarda ?

Comment (Med1233) [reply centreon/www/front_src/src/CloudNotificationsConfiguration/Listing/index.tsx:52]: and I will need it later for duplication and deletion

**Review by cg-tw — APPROVED** [review]

File: centreon/www/front_src/src/CloudNotificationsConfiguration/Listing/index.tsx (hunk 1 above)
Comment (Med1233) [reply centreon/www/front_src/src/CloudNotificationsConfiguration/Listing/index.tsx]: Done! @Thebarda

**Review by Thebarda — APPROVED** [review]

**Review by cg-tw — APPROVED** [review]

**Review by abuathier — APPROVED** [review]

**Pull request merged**